src-docs/charm_state.py.md

```diff
@@ -0,0 +1,103 @@
+<!-- markdownlint-disable -->
+
+<a href="../src/charm_state.py#L0"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+# <kbd>module</kbd> `charm_state.py`
+Module defining the CharmState class which represents the state of the SMTP Integrator charm. 
+
+**Global Variables**
+---------------
+- **KNOWN_CHARM_CONFIG**
+
+
+---
+
+## <kbd>class</kbd> `CharmConfigInvalidError`
+Exception raised when a charm configuration is found to be invalid. 
+
+Attrs:  msg (str): Explanation of the error. 
+
+<a href="../src/charm_state.py#L56"><img align="right" style="float:right;" src="https://img.shields.io/badge/-source-cccccc?style=flat-square"></a>
+
+### <kbd>function</kbd> `__init__`
+
+```python
+__init__(msg: str)
+```
+
+Initialize a new instance of the CharmConfigInvalidError exception. 
+
+
+
+**Args:**
+ 
+ - <b>`msg`</b> (str):  Explanation of the error. 
+
+
+
+
+
+---
+
+## <kbd>class</kbd> `CharmState`
+Represents the state of the SMTP Integrator charm. 
+
+Attrs:  host: The hostname or IP address of the outgoing SMTP relay.  port: The port of the outgoing SMTP relay.  user: The SMTP AUTH user to use for the outgoing SMTP relay.  password: The SMTP AUTH password to use for the outgoing SMTP relay.  auth_type: The type used to authenticate with the SMTP relay.  transport_security: The security protocol to use for the outgoing SMTP relay.  domain: The domain used by the sent emails from SMTP relay. 
```

Comment:
This needs to be called `Attributes` so that it renders properly here please

Reply:
Noted. Thanks